lib/src/third_party/diff_match_patch/diff.dart

```diff
@@ -0,0 +1,990 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:collection';
+import 'dart:math';
+
+import 'package:meta/meta.dart';
+
+part 'utils.dart';
+
+enum Operation { delete, insert, equal }
+
+/// Class representing one diff operation.
+class Diff {
+  /// One of: Operation.insert, Operation.delete or Operation.equal.
+  Operation operation;
+
+  /// The text associated with this diff operation.
+  String text;
+
+  /// Constructor.  Initializes the diff with the provided values.
+  ///
+  /// [operation] is one of Operation.insert, Operation.delete or Operation.equal.
+  /// [text] is the text being applied.
+  Diff(this.operation, this.text);
+
+  /// Display a human-readable version of this Diff.
+  /// Returns a text version.
+  @override
+  String toString() {
+    var prettyText = text.replaceAll('\n', '\u00b6');
+    return 'Diff($operation,"$prettyText")';
+  }
+
+  /// Is this Diff equivalent to another Diff?
+  /// [other] is another Diff to compare against.
+  /// Returns true or false.
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is Diff &&
+          runtimeType == other.runtimeType &&
+          operation == other.operation &&
+          text == other.text;
+
+  /// Generate a uniquely identifiable hashcode for this Diff.
+  /// Returns numeric hashcode.
+  @override
+  int get hashCode => operation.hashCode ^ text.hashCode;
+}
+
+class DiffMatchPatch {
+  double diffTimeout = 1.0;
+
+  /// Find the differences between two texts.
+  ///
+  /// Simplifies the problem by stripping any common prefix or suffix
+  /// off the texts before diffing.
+  /// [text1] is the old string to be diffed.
+  /// [text2] is the new string to be diffed.
+  /// [checklines] is an optional speedup flag.  If present and false, then don't
+  /// run a line-level diff first to identify the changed areas.
+  /// Defaults to true, which does a faster, slightly less optimal diff.
+  /// [deadline] is an optional time when the diff should be complete by.  Used
+  /// internally for recursive calls.  Users should set DiffTimeout instead.
+  /// Returns a List of Diff objects.
+  List<Diff> diffMain(
+    String text1,
+    String text2, {
+    bool checklines = true,
+    DateTime? deadline,
+  }) {
+    // Set a deadline by which time the diff must be complete.
+    if (deadline == null) {
+      deadline = DateTime.now();
+      if (diffTimeout <= 0) {
+        // One year should be sufficient for 'infinity'.
+        deadline = deadline.add(const Duration(days: 365));
+      } else {
+        deadline =
+            deadline.add(Duration(milliseconds: (diffTimeout * 1000).toInt()));
+      }
+    }
```

Comment:
Why don't we just make `deadline` a required argument, then we don't need `diffTimeout`.

Or we do `deadline ??= Datetime.now().add(const Duration(seconds: 1));` isn't that exactly the default anyways?